Introduce the notion of context

tidoust opened the pull request:
This update introduces the notion of a chunk context with wording and example taken from:
https://github.com/w3c/cogai/blob/master/chunks-and-rules.md#statements-about-statements

It also describes the `@context` property and updates the "matching chunk" algorithm accordingly (algorithm was re-worded to better account for other corner cases. It still does not account for variables and for the `~` operator).

Linked to #25.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/tidoust/cogai/pull/29.html" title="Last updated on Mar 23, 2021, 4:45 PM UTC (6701723)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/cogai/29/1f6ec06...tidoust:6701723.html" title="Last updated on Mar 23, 2021, 4:45 PM UTC (6701723)">Diff</a>

tidoust committed:
Introduce the notion of context

This update introduces the notion of a chunk context with wording and example
taken from:
https://github.com/w3c/cogai/blob/master/chunks-and-rules.md#statements-about-statements

It also describes the `@context` property and updates the "matching chunk"
algorithm accordingly (algorithm was re-worded to better account for other
corner cases. It still does not account for variables and for the `~` operator).

Linked to #25.

tidoust committed:
Contexts: re-order text and adjust wording slightly

draggett commented:
You could note that chunks with @context won't match chunks without @context since the latter belong to the default context.

Rule sets and graph algorithms can be used to reason across contexts, including the possibility of context chains, for example, when reasoning about the reported beliefs of someone in a work of fiction.

tidoust committed:
Explain the use of context for matching chunks

This adds some text that explain, in layman's terms, how contexts affect
matching rules.

tidoust commented:
> You could note that chunks with @context won't match chunks without @context since the latter belong to the default context.

Good point. I added a paragraph that clarifies how contexts affect chunk matching.
